pandas/core/groupby.py

```diff
@@ -1455,6 +1455,26 @@ def expanding(self, *args, **kwargs):
         from pandas.core.window import ExpandingGroupby
         return ExpandingGroupby(self, *args, **kwargs)
 
+    def _fill(self, how, limit=None):
+        labels, _, _ = self.grouper.group_info
+
+        # Need int value for Cython
+        if limit is None:
+            limit = -1
+        output = {}
+        if type(self) is DataFrameGroupBy:
```

Comment:
What's this if branch for?

Reply:
Fill is unique in that unlike say transformations the column(s) used in the grouping are also returned as part of the output of a DataFrame object. `_iterate_slices` does not go over those items and I didn't see an existing method to handle that, so this conditional makes sure the grouped columns still make their way into the frame that gets returned

Reply:
FWIW I was thinking to myself if its worth adding an instance method to the GroupBy object to handle this (`_iterate_groups` or something to the effect) but couldn't think of any other function that requires this feature

Reply:
rather than doing the ``if type(...)``, make a method in SeriesGroupBy and DataFrameGroupBy